sapan-parikh/SAPAN_PARIKH.md

```diff
@@ -1,16 +1,34 @@
+This discussion brings us back to the testing pyramid. Any organization or project's testing strategy should include a combination of unit tests, integration tests, Selenium tests, manual tests, etc.
 
-This discussion brings us back to the testing pyramid. Any organization or project'stesting strategy should include unit tests, integration tests, Selenium tests, manual tests, etc. Such atest strategy can tell us exactly the problems arewith minimum effort and time.This is how tests differ in size and speed, but they also differ in their impact and risk.It makes sense that when your project is small, you run all the tests, but as it grows, executing all tests may take hours or days. A simple strategy at such times is to distribute your test cases in a quadrant of the likelihoodof failure and impact of failure. A test case with a higher possibility of failure and a high impactisthe kindthat should be prioritized.
 
-Slow running test cases can be given score between 1 to 5 for their possibility of failure and impact of failure. Based on historical data, one can find the probability of the test case failing again. mpact can be based on the possible damage if feature work. The product of both these scores should give us a risk score. Which typically looks like the following:
+Image credit: https://martinfowler.com/articles/practical-test-pyramid.html
+
+Companies like Google tag the test cases by their sizes S, M, L, and XL, which I believe is the same thing as the test pyramid, wherein as you go towards the tip of the pyramid, your tests get bigger in terms of execution time as well as efforts.
+
+A test strategy that breaks down the test cases by size and/or type can efficiently tell us exactly where the problems are, with minimal effort and time.
+
+### Risk-based testing
+Now that we know that not all tests are the same in size and speed, we will explore how they also differ in their impact and risk. It makes sense to run all the tests when a project is small enough, but as it grows, executing all tests may take hours or days. A simple strategy at such times is to distribute your test cases in a quadrant of likelihood of failure and impact of failure. A test case with a higher possibility of failure and a high impact is the kind that should be prioritized.
+To use this methodology, we assign two measures to each of the test cases.
+Frequency of failure: Using historical data, one can figure out the likelihood of failure. A test that's always known to pass can be given the score of one, and a test that's know to fail a lot can get a score of five.
+Impact: A failing test case which does not harm much can be given a score of one. Versus, if there are scenarios that if broken, can do permanent damage to the company's image, then they should be given a score of five.
+This scoring finally should be arranged in a matrix like the following image.
 
 ![Slow running](sapan-parikh/slowrunning.png "Slow Running")
```

Comment:
"running" should be capitalised

Reply:
Is this supposed to be a header and a hyperlink rather than just a hyperlink?

Reply:
Was supposed to be an image.